src/chain/validators.ts

```diff
@@ -36,6 +36,14 @@ export interface Validators<Return> {
    */
   custom(validator: CustomValidator): Return;
 
+  /**
+   * Adds a field rename functionality to the validation chain.
+   *
+   * @param evaluator the custom evaluator based on the `CustomValidator`
+   * @returns the current validation chain
+   */
+  rename(evaluator: CustomValidator): Return;
```

Comment:
Nice, this is better!
A few comments here though:
1. This shouldn't stay in the `Validators` interface. `ContextHandler` sounds more like it.
2. Let's add a type specific for the rename callback, so that `CustomValidator` can evolve independently from the rename functionality.
3. WDYT of accepting a string value too, which works as a shortcut? For example, `rename('banana')` automatically renames that field to `banana`.